client/onboarding-prototype/types.ts

```diff
@@ -17,4 +17,17 @@ export type OnboardingFields = {
 	go_live_timeframe?: string;
 };
 
+export interface ProgressiveOnboardingEligibleResult {
+	result: string;
+	data: Array< string >;
```

Comment:
basically, data is returned as `[]` in a success case, but can contain some errors if unauthorized etc.
Is this a good type for that?

Reply:
If we're not going to handle those errors, I think it's okay to leave `data` out of the definition. In fact, it should be an object and not an array, but that's a server thing 🙂 

As result might not exist due to an error response, we should write it like optional, and use string literal types. Something like:
```ts
export interface EligibleResult {
	result: 'eligible' | 'not_eligible';
}
```

Reply:
Addressed in https://github.com/Automattic/woocommerce-payments/pull/5779/commits/81a0430e86b35d74a140e4916e0368747bfff8b3

Reply:
Oh, I need to remove `data: Array< string >;`, will do it as well!

Reply:
Done in https://github.com/Automattic/woocommerce-payments/pull/5779/commits/28e46d6ff48a2135b97eb10d68b6052c48b33305